src/main/java/de/tum/in/www1/artemis/repository/metis/conversation/OneToOneChatRepository.java

```diff
@@ -56,16 +56,24 @@ public interface OneToOneChatRepository extends JpaRepository<OneToOneChat, Long
     @Query("""
             SELECT DISTINCT o
             FROM OneToOneChat o
-                LEFT JOIN FETCH o.conversationParticipants p1
-                LEFT JOIN FETCH o.conversationParticipants p2
-                LEFT JOIN FETCH p1.user u1
-                LEFT JOIN FETCH p2.user u2
-                LEFT JOIN FETCH u1.groups
-                LEFT JOIN FETCH u2.groups
+                LEFT JOIN FETCH o.conversationParticipants p
+                LEFT JOIN FETCH p.user u
+                LEFT JOIN FETCH u.groups
             WHERE o.course.id = :courseId
-                AND u1.id = :userIdA
-                AND u2.id = :userIdB
+                AND EXISTS (
+                    SELECT 1
+                    FROM ConversationParticipant cp
+                    WHERE cp.conversation = o
+                        AND cp.user.id = :userIdA
+                )
+                AND EXISTS (
+                    SELECT 1
+                    FROM ConversationParticipant cp
+                    WHERE cp.conversation = o
+                        AND cp.user.id = :userIdB
+                )
```

Comment:
Would `AND u.id IN (:userIdA, :userIdB)` work?

Reply:
No, that would return any that contain one of those.

That would in Java terms be like

Required: `participantIds.equals(List.of(userIdA, userIdB));`
Suggested: `participantIds.containsAll(List.of(userIdA, userIdB));`

Reply:
And because of the join, that would then return many more results.